### ISSUE TITLE
F12 (Go to definition) doesn't work in a contextual sense

### ISSUE THREAD

**Issue by rossmkynch** [hpcc-systems/vscode-ecl#350]
Contextual "go to definition" (F12 shortcut) doesn't work for ECL. There are two main ways this manifests:

1: Files within the same folder.
ECL files within the same folder are able to reference each other directly, without need for qualifying scope. i.e. SomeFolder/File2.ecl may use SomeFolder/File1.ecl's functions directly, without the need for IMPORT SomeFolder or using SomeFolder.File1.someFunction. i.e. you can use File1.someFunction, but go to definition/F12 does not navigate to the file in VSC.

2: Nested folder imports.
You may use IMPORT SomeFolder.NestedFolder to access functionality. Calls must then be made using NestedFolder.SomeFunction. However, go to definition/F12 does not navigate to the file in VSC.

There is also the IMPORT SomeFolder AS SF, then calls using SF.function will also fail to go to definition, although this is more complex and understandable behaviour.

**Comment by rossmkynch** [comment]
Would like this addressed, if possible

**Comment by GordonSmith** [comment]
Are you using the legacy flag?

**Comment by rossmkynch** [comment]
> Are you using the legacy flag?

Yes, will try with it off, although I think I enabled it to solve other problems
